src/snapred/backend/data/GroceryService.py

```diff
@@ -865,13 +863,17 @@ def fetchGroupingDefinition(self, item: GroceryListItem) -> Dict[str, Any]:
 
         :rtype: Dict[str, Any]
         """
-        key = self._key(item.groupingScheme, item.runNumber, item.useLiteMode)
+        if item.runNumber in ReservedRunNumber.values():
+            stateId = ReservedStateId.forRun(item.runNumber)
+        else:
+            stateId, _ = self.dataService.generateStateId(item.runNumber)
```

Comment:
Question for reviewer: Should this got here, or inside `LocalDataService.generateStateId`?